### PR TITLE
Add basic documentation

### PR DESCRIPTION
**Pull request by Ericgig**
Basic documentation structure.
- Installation information
- Example of auto-differentiation
- Re-active github actions

### CODE REVIEW

**Comment by Ericgig** [comment]
The failing test are fixed in #19

File: doc/source/basics.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+************
+
+
+.. _basic_usage:
+
+Basic Usages
+============
+
+Once imported, jax array will be available to contain ``qutip.Qobj``'s data and diffrax ODE will be available for solver (``sesolve``, ``mcsolve``, etc.).
+None of the function in the module are expected to be used directly, they will be used by qutip when the options to use this backend is used.
```
Comment (AGaliciaMartinez) [inline doc/source/basics.rst:14]:
```suggestion
In orther to enable qutip-jax, it is just necessary to import the module. Once imported, ``qutip.Qobj``'s data can be represented as a JAX array. Furthermore, diffrax ODE will be available as an option for qutip's solvers (``sesolve``, ``mcsolve``, etc.).
None of the functions in the module are expected to be used directly. Instead, they will be used by qutip, allowing the user to interact only with the already familiar QuTiP interface.
```

File: doc/source/basics.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+There are many ways to tell Qutip to use the jax backend.
```
Comment (AGaliciaMartinez) [inline doc/source/basics.rst:16]:
```suggestion
There are many ways to create a QuTiP ``Qobj`` backed by JAX's array class.
```

File: doc/source/basics.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+- Pass Jax array to the Qobj constructor.
```
Comment (AGaliciaMartinez) [inline doc/source/basics.rst:18]:
```suggestion
- Passing JAX array to the ``Qobj`` constructor.
```

File: doc/source/basics.rst [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+.. code-block::
+
+    import qutip
+    import qutip_jax
+    import jax.numpy as jnp
+
+    jax_eye_qobj = qutip.Qobj(jnp.eye(3))
+
+- Manually convert a ``Qobj`` or ``QobjEvo``.
```
Comment (AGaliciaMartinez) [inline doc/source/basics.rst:28]:
```suggestion
- Converting a ``Qobj`` or ``QobjEvo`` using the ``to`` method.
```

File: doc/source/basics.rst [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+- Pass Jax array to the Qobj constructor.
+
+.. code-block::
+
+    import qutip
+    import qutip_jax
+    import jax.numpy as jnp
+
+    jax_eye_qobj = qutip.Qobj(jnp.eye(3))
```
Comment (AGaliciaMartinez) [inline doc/source/basics.rst:26]: Perhaps in these codeblocks we can add something at the end (`jax_eye_qobj.data` for example) to make it easier to whoever is executing this code what to expect or how to assess it jax is being used.

File: doc/source/basics.rst [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+- Manually convert a ``Qobj`` or ``QobjEvo``.
+
+.. code-block::
+
+    import qutip
+    import qutip_jax
+
+    jax_eye_qobj = qutip.qeye(3).to("jax")
+
+- Use qutip state and operator function with ``dtype="jax"``.
```
Comment (AGaliciaMartinez) [inline doc/source/basics.rst:37]:
```suggestion
- Using QuTiP's native constructors' ``dtype`` parameter.
```

File: doc/source/basics.rst [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+.. code-block::
+
+    import qutip
+    import qutip_jax
+
+    jax_eye_qobj = qutip.qeye(3, dtype="jax")
+
+- Use qutip settings to set jax as the default backend.
```
Comment (AGaliciaMartinez) [inline doc/source/basics.rst:46]:
```suggestion
- Changing QuTiP's default settings to set JAX as the default backend.
```

File: doc/source/basics.rst [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+.. code-block::
+
+    import qutip
+    import qutip_jax
+    qutip.settings.core["default_dtype"] = "jax"
+
+    jax_eye_qobj = qutip.qeye(3)
+
+- Use qutip settings in a context.
```
Comment (AGaliciaMartinez) [inline doc/source/basics.rst:56]:
```suggestion
- Changing QuTiP's default settings within a context.
```

File: doc/source/installation.rst [hunk 9]
```diff
@@ -0,0 +1,33 @@
+************
+Installation
+************
+
+.. _quickstart:
+
+Qutip-Jax needs the development version of Qutip and installation is only possible from source
```
Comment (AGaliciaMartinez) [inline doc/source/installation.rst:7]:
```suggestion
Qutip-Jax needs the development version of Qutip and installation is only possible from source.
```

File: doc/source/installation.rst [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+    matplotlib pytest
+
+In addition
+
+.. code-block:: bash
+
+    sphinx numpydoc sphinx sphinx_rtd_theme
+
+are used to build and test the documentation.
```
Comment (AGaliciaMartinez) [inline doc/source/installation.rst:33]: Minor thing: This perhaps  suggest that these packages need to be installed manually yet the testing requirements should be in setup.cfg and the documentation requirements and in doc/requirementes.txt.

File: doc/source/introduction.rst [hunk 11]
```diff
@@ -0,0 +1,9 @@
+************
+Introduction
+************
+
+The qutip-jax package
+=====================
+
+This package add the capacity to use jax Array in ``qutip.Qobj``.
```
Comment (AGaliciaMartinez) [inline doc/source/introduction.rst:8]:
```suggestion
This package adds the capacity to use JAX Array in ``qutip.Qobj``.
```

File: doc/source/introduction.rst [hunk 12]
```diff
@@ -0,0 +1,9 @@
...
+This allow to use Jax's auto differentiation and device (gpu) support in Qutip.
```
Comment (AGaliciaMartinez) [inline doc/source/introduction.rst:9]:
```suggestion
This allows using Jax's auto differentiation and device (gpu) support in Qutip.
```

File: doc/source/solver.rst [hunk 13]
```diff
@@ -0,0 +1,31 @@
+.. _qtjax_solver:
+
+*****************************
+Jax BackEnd in Qutip's Solver
```
Comment (AGaliciaMartinez) [inline doc/source/solver.rst:4]:
```suggestion
Jax Backend in Qutip's Solver
```

File: doc/source/solver.rst [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+*****************************
+
+
+.. _mesolve:
+
+Using Jax in ``mesolve``
+========================
+
+
+To use the jax arrays in Qutip solver such as ``mesolve``, two conditions must be fullfill:
```
Comment (AGaliciaMartinez) [inline doc/source/solver.rst:14]:
```suggestion
To use JAX's arrays in QuTiP's solvers such as ``mesolve``, two conditions must be met:
```

File: doc/source/solver.rst [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+
+1. All opertors and states must use jax arrays.
```
Comment (AGaliciaMartinez) [inline doc/source/solver.rst:16]:
```suggestion
1. All opertors and states must use JAX arrays.
```

File: doc/source/solver.rst [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+2. An ODE integrator supporting Jax must be used.
+   ODE solver from the diffrax project are made available when importing qutip-jax
```
Comment (AGaliciaMartinez) [inline doc/source/solver.rst:18]:
```suggestion
2. An ODE integrator supporting Jax must be used.
   Currently, an ODE solver from the diffrax project is made available when importing qutip-jax.
   
The following code shows an example of how to use JAX:
```

I remember you mentioned that JAX  was a bit slower the first time it is executed. Is this always the case? If so, perhaps we should briefly explain here why this happens.

**Review by AGaliciaMartinez — CHANGES_REQUESTED** [review]
Great work @Ericgig sumarizing the current features of qutip-jax! I made some suggestions hoping to improve what you wrote in the documentation. However, feel free to reject them if you prefer the original version. I would like to reread the documentation again once the changes are included. I still would like to compile it locoally to ensure everything reders correctly.

File: doc/source/solver.rst (hunk 16 above)
Comment (Ericgig) [reply doc/source/solver.rst:18]: The first time compile the functions. It's true for any `jit` function, but the compilation is quite quick.

File: doc/source/installation.rst (hunk 10 above)
Comment (Ericgig) [inline doc/source/installation.rst:33]: I will simplify:
```suggestion
We use ``pytest`` for testing.

See doc/requirementes.txt for a list of package needed to build the documentation.
```

File: doc/source/basics.rst (hunk 5 above)
Comment (Ericgig) [reply doc/source/basics.rst:26]: Done, was it what you had in mind?

Comment (AGaliciaMartinez) [reply doc/source/basics.rst:26]: Yes, looks great now.

**Comment by AGaliciaMartinez** [comment]
Thanks for applying the changes, I will do one more proof read and then we can merge.

File: doc/source/index.rst [hunk 17]
```diff
@@ -0,0 +1,48 @@
+.. qutip_jax
+
+.. figure:: figures/logo.png
+    :align: center
+    :width: 5in
+
+*********************************************************
+qutip-jax |version|: Jax-backend for QuTiP
```
Comment (AGaliciaMartinez) [inline doc/source/index.rst:8]: Is this part being rendered properly?  The duplication of the title looks a bit odd.
image

File: doc/source/basics.rst [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+
+************
+Jax in Qutip
+************
+
+
+.. _basic_usage:
+
+Basic Usages
+============
```
Comment (AGaliciaMartinez) [inline doc/source/basics.rst:11]:
```suggestion
Basic usage
===========
```

File: doc/source/basics.rst [hunk 19]
```diff
@@ -0,0 +1,70 @@
+.. _qtjax_intro:
+
+************
+Jax in Qutip
```
Comment (AGaliciaMartinez) [inline doc/source/basics.rst:4]:
```suggestion
Jax in QuTiP
```

File: doc/source/solver.rst [hunk 20]
```diff
@@ -0,0 +1,33 @@
+.. _qtjax_solver:
+
+*****************************
+Jax Backend in Qutip's Solver
```
Comment (AGaliciaMartinez) [inline doc/source/solver.rst:4]:
```suggestion
Jax Backend in QuTiP's Solver
```

**Review by AGaliciaMartinez — APPROVED** [review]

File: doc/source/index.rst [hunk 21]
```diff
@@ -0,0 +1,48 @@
...
+.. figure:: figures/logo.png
+    :align: center
+    :width: 5in
+
+*********************************************************
+qutip-jax |version|: Jax-backend for QuTiP 
+*********************************************************
+
+qutip-jax: JAX backend for QuTiP
+==============================================
```
Comment (Ericgig) [inline doc/source/index.rst:12]:
```suggestion
```

File: doc/source/index.rst (hunk 17 above)
Comment (Ericgig) [reply doc/source/index.rst:8]: Removed the duplicate.

**Pull request merged**